### ISSUE TITLE
use __getattr__ and return self doesn't allow method chaining

### ISSUE THREAD

**Issue by pjbonestroo** [QQuick/Transcrypt#587]
I use `__getattr__` and `__setattr__` to wrap an html element. Then I'm able to set properties on the html element using the wrapper. For this example, the html element is replaced by ` class Element` to let it work in CPython.

It all works nice, but only the method chaining doesn't work:

```
class Element():
    def __init__(self):
        self.message = "Goodbye"

    def sayBye(self):
        print(self.message)

class Wrapper():
    def __init__ (self, element):
        self.element = element

    def __setattr__ (self, name, value):
        """ set attribute on element if it already has the attribute """
        if name != "element" and hasattr(self.element, name):
            setattr(self.element, name, value)
        else:
            self.__dict__[name] = value

    def __getattr__ (self, name):
        """ get attribute from element if this object doesn't have the attribute """
        result = getattr(self.element, name)
        # if result is a function, bind self.element to it
        if hasattr(result, 'call') and hasattr(result, 'bind'):
            result = result.bind(self.element)
        return result

    def sayHello(self):
        print("Hello")
        return self


e = Element()
w = Wrapper(e)

#
# Usage
#

w.message = "Bye" # set attributes on e, using w
e.sayBye()
w.sayBye() # call functions on e, using w

# and method chaining should work:
w.sayHello().sayBye()
```

Returning self in method `sayHello` gives the `Wrapper` object, but not the javascript Proxy object. Is this intentional, and thus a restriction (it works in CPython)?
Or is it a bug?

**Comment by pjbonestroo** [comment]
Hi Jacques,
I hope it can be done while maintaining performance. I didn't think about a solution yet. If I find time, I will look into it.
Thanks in advance for helping.
Regards, Pieter Jan

**Comment by pjbonestroo** [comment]
A trial in the generated javascript would be like:

org.transcrypt.__runtime__.js
```
export function __get__ (self, func, quotedFuncName) {
    if (self) {
        if (self.hasOwnProperty ('__class__') || typeof self == 'string' || self instanceof String) {
            if (quotedFuncName) {
                Object.defineProperty (self, quotedFuncName, {
                    value: function () {
                        var args = [] .slice.apply (arguments);
                        return func.apply (null, [self] .concat (args));
                    },
                    writable: true,
                    enumerable: true,
                    configurable: true
                });
            }
            return function () {
                var args = [] .slice.apply (arguments);
                if (self.__proxy__ != undefined){
                    return func.apply (null, [self.__proxy__] .concat (args));
                }else{
                    return func.apply (null, [self] .concat (args));
                }
            };
        }
        else {
            return func;
        }
    }
    else {
        return func;
    }
};
...
...
export var object = {
    __init__: function (self) {},
    __metaclass__: py_metatype,
    __name__: 'object',
    __bases__: [],
    __new__: function (args) {
        var instance = Object.create (this, {__class__: {value: this, enumerable: true}});
        if ('__getattr__' in this || '__setattr__' in this) {
            var inst = new Proxy (instance, {
                get: function (target, name) {
                    let result = target [name];
                    if (result == undefined) {
                        return target.__getattr__ (name);
                    }
                    else {
                        return result;
                    }
                },
                set: function (target, name, value) {
                    try {
                        target.__setattr__ (name, value);
                    }
                    catch (exception) {
                        target [name] = value;
                    }
                    return true;
                }
            })
            instance.__proxy__ = inst;
            instance = inst;
        }
        this.__init__.apply (null, [instance] .concat (args));
        return instance;
    }
};
```

Each instance now has a __proxy__ attribute.
I'm not sure if this is the best solution.
Best regards

**Comment by JdeH** [comment]
Hi Pieter Jan

I've manually merged your solution, see #591, and added an autotest to
the transcrypt/proxies testlet, based on your code.

Kind regards,
Jacques
